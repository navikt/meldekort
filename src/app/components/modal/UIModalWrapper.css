.uimodal {
  width: 40rem;
  min-height: 15rem;
  position: relative;
}
.uimodal__content {
  margin-top: 1rem;
}
.uimodal__actions {
  position: absolute;
  bottom: 1rem;
}
.uimodal__actions-knapp {
  margin-left: 1rem;
}
.uimodal .uimodal__actions-knapp {
  margin-top: 0.5rem;
}
.modal-overskrift h2 {
  margin-bottom: 1rem;
}
