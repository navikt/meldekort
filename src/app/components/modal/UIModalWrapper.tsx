import * as React from 'react';
import { IModal, ModalKnapp } from '../../types/ui';
import { RootState } from '../../store/configureStore';
import { selectModal } from '../../selectors/ui';
import { Dispatch } from 'redux';
import { UiActions } from '../../actions/ui';
import { connect } from 'react-redux';
import KnappBase from 'nav-frontend-knapper';
import Modal from 'nav-frontend-modal';

interface MapStateToProps {
  modal: IModal;
}

interface MapDispatchToProps {
  skjulModal: () => void;
}

type UIModalWrapperProps = MapStateToProps & MapDispatchToProps;

const UIModalWrapper: React.FunctionComponent<UIModalWrapperProps> = ({
  modal,
  skjulModal
}) => {
  Modal.setAppElement('#meldekort__root');
  return (
    <Modal
      className={'uimodal'}
      isOpen={modal.visModal}
      onRequestClose={() => {
        modal.onRequestClose ? modal.onRequestClose() : skjulModal();
      }}
      contentLabel={'ui-modal'}
      closeButton={true}
    >
      <div className={'uimodal__content'}>{modal.content()}</div>
      <div className={'uimodal__actions-knapp'}>
        {modal.knapper &&
          modal.knapper.map((knapp: ModalKnapp) => {
            return (
              <KnappBase
                key={knapp.label}
                className={'uimodal__actions-knapp'}
                type={knapp.type}
                onClick={() => knapp.action()}
              >
                {knapp.label}
              </KnappBase>
            );
          })}
      </div>
    </Modal>
  );
};

const mapStateToProps = (state: RootState): MapStateToProps => {
  return {
    modal: selectModal(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): MapDispatchToProps => {
  return {
    skjulModal: () => dispatch(UiActions.skjulModal()),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UIModalWrapper);
