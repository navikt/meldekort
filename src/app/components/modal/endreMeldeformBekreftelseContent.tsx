import { Normaltekst, Undertittel } from 'nav-frontend-typografi';
import { FormattedMessage } from 'react-intl';
import veileder from '../../ikoner/veileder.svg';
import Veilederpanel from 'nav-frontend-veilederpanel';
import * as React from 'react';
import { hentIntl } from '../../utils/intlUtil';

export const endreMeldeformBekreftelseContent = () => {
  return (
    <div className={'modal-overskrift'}>
      <Undertittel
        children={<FormattedMessage id={'overskrift.bekreftelse'} />}
      />
      <Veilederpanel fargetema="suksess" svg={<img alt="" src={veileder} />}>
        <div className="modal-tekst">
          <Normaltekst>
            {`${hentIntl()
              .formatMessage({ id: 'endreMeldeform.valgtBytte' })
              .trim()}
                            ${hentIntl().formatMessage({
                              id: 'endreMeldeform.elektronisk',
                            })}`}
          </Normaltekst>
          <Normaltekst>
            <FormattedMessage
              id={'endreMeldeform.kvittering.elektronisk.info'}
            />
          </Normaltekst>
        </div>
      </Veilederpanel>
    </div>
  );
};
