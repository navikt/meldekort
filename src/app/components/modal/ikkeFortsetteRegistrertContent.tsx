import { Normaltekst, Undertittel } from "nav-frontend-typografi";
import * as React from "react";
import { formatHtmlMessage } from "../../utils/intlUtil";
import { Alert } from "@navikt/ds-react";

export const ikkeFortsetteRegistrertContent = () => {
  return (
    <div className="modal-overskrift">
      <Undertittel>{formatHtmlMessage("sporsmal.bekreft")}</Undertittel>
      <Alert variant="warning">
        <div className="modal-tekst">
          <Normaltekst>
            {formatHtmlMessage("sporsmal.bekreftelse")}
          </Normaltekst>
        </div>
      </Alert>
    </div>
  );
};
