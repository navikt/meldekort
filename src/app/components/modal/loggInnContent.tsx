import { Normaltekst, Undertittel } from 'nav-frontend-typografi';
import * as React from 'react';
import Veilederpanel from 'nav-frontend-veilederpanel';
import veilederAdvarsel from '../../ikoner/veileder_advarsel.svg';

export const loggInnContent = () => {
    return (
        <div className="modal-overskrift">
            <Undertittel children={'Du må logge inn'}/>
            <Veilederpanel fargetema="advarsel" svg={<img src={veilederAdvarsel}/>}>
                <div className="modal-tekst">
                <Normaltekst>
                    Du er ikke logget inn. Vi viderefører deg til innloggingsiden.
                </Normaltekst>
                </div>
            </Veilederpanel>
        </div>
    );
};