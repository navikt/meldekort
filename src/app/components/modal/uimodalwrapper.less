.uimodal {
  width: 40rem;
  min-height: 15rem;
  position: relative;

  &__content {
    margin-top: 1rem;
  }

  &__actions {
    position: absolute;
    bottom: 1rem;

    &-knapp {
      margin-left: 1rem;
    }
  }
}

.modal-overskrift {
  h2 {
    margin-bottom: 1rem;
  }
}