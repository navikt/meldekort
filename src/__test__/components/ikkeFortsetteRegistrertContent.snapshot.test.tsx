import * as React from 'react';
import { ikkeFortsetteRegistrertContent } from '../../app/components/modal/ikkeFortsetteRegistrertContent';

it('Sprakvelger', () => {
  expect(ikkeFortsetteRegistrertContent()).toMatchSnapshot();
});
