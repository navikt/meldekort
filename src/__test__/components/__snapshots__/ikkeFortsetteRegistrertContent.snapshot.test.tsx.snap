// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ikkeFortsetteRegistrertContent 1`] = `
<div
  className="modal-overskrift"
>
  <Undertittel>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "sporsmal.bekreft",
        }
      }
    />
  </Undertittel>
  <ForwardRef
    variant="warning"
  >
    <div
      className="modal-tekst"
    >
      <Normaltekst>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "sporsmal.bekreftelse",
            }
          }
        />
      </Normaltekst>
    </div>
  </ForwardRef>
</div>
`;
